src/components/logs/logsWebview.ts

```diff
@@ -0,0 +1,145 @@
+
+export class LogsPanel {
+    public static readonly viewType = 'vscodeKubernetesLogs';
+    public static currentPanels = [];
+
+    private disposables: vscode.Disposable[] = [];
+    private content: string;
+    private resource: string;
+
+    public static createOrShow(extensionPath: string, content: string, resource: string): LogsPanel {
+        const column = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+        // If we already have a panel, show it.
+        const currentPanel = LogsPanel.currentPanels[resource];
+        if (currentPanel) {
+            currentPanel.setInfo(content, resource);
+            currentPanel.update();
+            currentPanel.panel.reveal(column);
+            return currentPanel;
+        }
+        const panel = vscode.window.createWebviewPanel(LogsPanel.viewType, "Kubernetes Logs", column || vscode.ViewColumn.One, {
+            enableScripts: true,
+
+            // And restrict the webview to only loading content from our extension's `media` directory.
+            localResourceRoots: [
+            ],
+        });
+
+        const result = new LogsPanel(panel, content, resource);
+        LogsPanel.currentPanels[resource] = result;
+        return result;
+    }
+
+    private constructor(
+        private readonly panel: vscode.WebviewPanel,
+        content: string,
+        resource: string
+    ) {
+        this.content = content;
+        this.resource = resource;
+
+        this.update();
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
+
+        this.panel.onDidChangeViewState((e: any) => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, null, this.disposables);
+    }
+
+    public setInfo(content: string, resource: string) {
+        this.content = content;
+        this.resource = resource;
+        this.update();
+    }
+
+    public dispose() {
+        delete LogsPanel.currentPanels[this.resource];
+
+        this.panel.dispose();
+
+        while (this.disposables.length) {
+            const x = this.disposables.pop();
+            if (x) {
+                x.dispose();
+            }
+        }
+    }
+
+    private update() {
+        this.panel.title = `Kubernetes logs ${this.resource}`;
+        this.panel.webview.html = `
+        <!doctype html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <title>Kubernetes logs ${this.resource}</title>
+            <script
+                src="https://code.jquery.com/jquery-3.3.1.min.js"
+                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
+                crossorigin="anonymous"></script>
+        </head>
+        <body>
+            <div style='position: absolute; top: 15px; left: 75%'>
+                <select id='mode' style='margin-bottom: 5px' onchange='eval()'>
+                    <option value='include'>Show Matches</option>
```

Comment:
Looking at your screenshots I worry this could overlap the first row of logs.  Another possible design would be toolbar style, with a bar at the top of the window reserved for these drop-downs, and the logs occupying the rest of the window a la:

```
+-------------------------------------------------+
| Show log entries: [...]  Match pattern: [...]   |
+-------------------------------------------------+
| 28.01.19 14:08:17 Printed l33t ASCII art        |
| ...                                             |
```

Reply:
done.

Reply:
_shakes head at HTML_